server/modules/routing/readwritesplit/rwsplit_session_cmd.cc

```diff
@@ -122,7 +122,7 @@ void process_sescmd_response(RWSplitSession* rses, SRWBackend& backend,
                     {
                         /** Map the returned response to the internal ID */
                         MXS_INFO("PS ID %u maps to internal ID %lu", resp.id, id);
-                        rses->ps_handles[resp.id] = id;
+                        rses->ps_handles[resp.id] = (id << 16) + resp.parameters;
```

Comment:
Left shift should be 32 bits since the stored value is 64 bits and the ID is 32.